homeassistant/components/assist_pipeline/websocket_api.py

```diff
@@ -287,6 +288,36 @@ def websocket_list_runs(
     )
 
 
+@callback
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "assist_pipeline/device/list",
+    }
+)
+def websocket_list_devices(
+    hass: HomeAssistant,
+    connection: websocket_api.connection.ActiveConnection,
+    msg: dict[str, Any],
+) -> None:
+    """List assist devices."""
+    pipeline_data: PipelineData = hass.data[DOMAIN]
+    ent_reg = er.async_get(hass)
+    devices: list[dict[str, Any]] = []
+
+    for device_id, info in pipeline_data.pipeline_devices.items():
+        devices.append(
+            {
+                "device_id": device_id,
+                "pipeline_entity": ent_reg.async_get_entity_id(
+                    "select", info.domain, f"{info.unique_id_prefix}-pipeline"
+                ),
+            }
+        )
```

Comment:
List comp would be slightly faster since we are building py3.12 now and they are inlined.

But there aren't enough of these to matter.